internal/api/mfa.go

```diff
@@ -71,14 +75,25 @@ func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
 	if session == nil || user == nil {
 		return internalServerError("A valid session and a registered user are required to enroll a factor")
 	}
-
 	params := &EnrollFactorParams{}
 	if err := retrieveRequestParams(r, params); err != nil {
 		return err
 	}
 
-	if params.FactorType != models.TOTP {
-		return badRequestError(ErrorCodeValidationFailed, "factor_type needs to be totp")
+	if params.FactorType != models.TOTP && params.FactorType != models.SMS {
+		return badRequestError(ErrorCodeValidationFailed, "factor_type needs to be totp or sms")
+	}
+	var phone string
+	var err error
+	if params.FactorType == models.SMS {
+		if !config.MFA.SMS.EnrollEnabled {
+			// TODO: Find proper error code
+			return badRequestError(ErrorCodeValidationFailed, "mfa enrolment is not enabled")
```

Comment:
This should be an unprocessable entity error with a new error code that signals MFA factor type not enabled.